### PR TITLE
ci-secret-bootstrap: skip secret sync on error

### PR DESCRIPTION
**Pull request by danilo-gemoli**
`ci-secret-bootstrap` fetches the secrets from the Vault before syncing them on a cluster.
Prior to this changeset, the `ci-secret-bootstrap` secrets fetching procedure was done at best effort, and it would potentially lead to data inconsistency/corruption.
What follows is a real use case we faced and that caused a serious outage on our CI (kind of, I'll make it up a bit for the sake of the explanation):
- we want to synchronize a secret in accordance with the following specification:
```yaml
secret_configs:
- from:
    sa.prow-controller-manager.app.ci.config:
      field: sa.prow-controller-manager.app.ci.config
      item: build_farm
    sa.prow-controller-manager.app.ci.token.txt:
      field: sa.prow-controller-manager.app.ci.token.txt
      item: build_farm
    sa.prow-controller-manager.build01.config:
      field: sa.prow-controller-manager.build01.config
      item: build_farm
    sa.prow-controller-manager.build01.token.txt:
      field: sa.prow-controller-manager.build01.token.txt
      item: build_farm
  to:
  - cluster: app.ci
    name: prow-controller-manager
    namespace: ci
```
- the Vault returns an error when queried on these keys: `sa.prow-controller-manager.build01.config`, `sa.prow-controller-manager.build01.token.txt`
- `ci-secret-bootstrap` reports and logs the error but it keeps going anyway and it finally writes a partially fetched secret on `app.ci`:
```yaml
kind: Secret
apiVersion: v1
metadata:
  labels:
    dptp.openshift.io/requester: ci-secret-bootstrap
  name: prow-controller-manager
  namespace: ci
type: Opaque
data:
  sa.prow-controller-manager.app.ci.config: <REDACTED>
  sa.prow-controller-manager.app.ci.token.txt: <REDACTED>
```
- we have in fact synchronized a partial secret

This PR prevents that scenario by skipping a secret when:
- the Vault reports an error
- the base64 decoding procedure fails. This wasn't the cause during the outage but I guess it makes sense to take it into account as it would cause troubles anyway.

/cc @jmguzik @jupierce

### CODE REVIEW

**Review by jmguzik — APPROVED** [review]
atomic, nice

File: cmd/ci-secret-bootstrap/main.go [hunk 1]
```diff
@@ -367,6 +371,12 @@ func constructSecrets(config secretbootstrap.Config, client secrets.ReadOnlyClie
 			// that we need to wait for that map to be fully populated.
 			keyWg.Wait()
 
+			// We don't want to sync secrets that have not been fully fetched from the secret manager
+			// and/or have not been properly constructed.
+			if secretInError.Load() {
+				return
```
Comment (jupierce) [inline cmd/ci-secret-bootstrap/main.go:377]: Maybe an explicit log entry here to say it is being skipped? Nice for someone reading the logs to have a breadcrumb know code was skipped.

**Review by jmguzik — APPROVED** [review]

**Comment by danilo-gemoli** [comment]
Alright the `e2e` is kind of broken but the failure is unrelated to this PR. I'll take a look into it but it's important we merge this fix.
/override ci/prow/e2e

**Pull request merged**